dmoj/executors/filesystem_rules.py

```diff
@@ -0,0 +1,23 @@
+class FilesystemPolicy:
+    def __init__(self, *items):
+        self.items = list(items)
+
+    def __add__(self, other):
+        return FilesystemPolicy(*(self.items + other.items))
+
+    def check(self, path):
+        return any(item.check(path) for item in self.items)
+
+class Dir:
+    def __init__(self, path):
+        self.path = path
+
+    def check(self, path):
+        return path.startswith(self.path)
```

Comment:
Seems like a security issue? Allowing `/usr` would end up allowing access to `/usr2/stuff`.

Reply:
can I fix this with something as trivial as `path.startswith(self.path + "/")` or should we do something more intelligent

Reply:
Appending / will break stat on the directory itself, so you'll need something more intelligent.

Reply:
I guess the next dumb idea is `path == self.path or path.startswith(self.path + "/")`

Reply:
You probably want to compare `os.path.split(path)` with `os.path.split(self.path)`. Easier to think about per-token comparison.

Reply:
That sounds much more sane

Reply:
What did the regex system do?

Reply:
It specified the directory like `/usr$` and the full match `/usr/`, once each?

Reply:
It never gave `/usr/`, t gave `/usr$` and `/usr/(?:home)`.

Reply:
Each rule has an implicit `^` at the start. The rule would have excluded home2, should such a user have existed. That's just an oversight.